src/lib/containers/QueryWrapper.tsx

```diff
@@ -277,13 +282,20 @@ export default class QueryWrapper extends React.Component<
       this.updateParentState,
     )
       .then((data: QueryResultBundle) => {
+        const isFaceted = isFacetAvailable(data.facets)
```

Comment:
When props change and query is re-executed, we update the isFaceted state.
Also update the top level control state (ability to show facet filters and visualizations)